native-pkcs11-core/src/mechanism.rs

```diff
@@ -29,25 +36,78 @@ pub enum Mechanism {
     RsaPkcsSha384,
     RsaPkcsSha512,
     RsaX509,
+    RsaPss {
+        digest_algorithm: native_pkcs11_traits::DigestType,
+        mask_generation_function: native_pkcs11_traits::DigestType,
+        salt_length: u64,
+    },
 }
 
-pub static MECHANISMS: Lazy<Vec<CK_MECHANISM_TYPE>> =
-    Lazy::new(|| Mechanism::iter().map(|m| m.into()).collect());
+pub unsafe fn parse_mechanism(mechanism: CK_MECHANISM) -> Result<Mechanism, Error> {
+    match mechanism.mechanism {
+        CKM_ECDSA => Ok(Mechanism::Ecdsa),
+        CKM_RSA_PKCS => Ok(Mechanism::RsaPkcs),
+        CKM_RSA_X_509 => Ok(Mechanism::RsaX509),
+        CKM_SHA1_RSA_PKCS => Ok(Mechanism::RsaPkcsSha1),
+        CKM_SHA256_RSA_PKCS => Ok(Mechanism::RsaPkcsSha256),
+        CKM_SHA384_RSA_PKCS => Ok(Mechanism::RsaPkcsSha384),
+        CKM_SHA512_RSA_PKCS => Ok(Mechanism::RsaPkcsSha512),
+        CKM_RSA_PKCS_PSS => {
+            //  Bind to locals to prevent unaligned reads https://github.com/rust-lang/rust/issues/82523
+            let mechanism_type = mechanism.mechanism;
+            let parameter_ptr = mechanism.pParameter;
+            let parameter_len = mechanism.ulParameterLen;
+            if parameter_ptr.is_null() {
+                tracing::error!("pParameter null");
+                return Err(Error::MechanismInvalid(mechanism_type));
+            }
+            if (parameter_len as usize) < std::mem::size_of::<CK_RSA_PKCS_PSS_PARAMS>() {
```

Comment:
What if the parameter_len is larger then size_of::<CK_RSA_PKCS_PSS_PARAMS>? Is that ok?

Should the code just check equality of the sizes?

Reply:
The thought is that if new fields are added to the struct in a later version it wouldn't break already-compiled versions of the module. I'm open to discussion on this. The probability of PSS params changing may be so low that we should just check equality anyways

Reply:
I changed this to `==` in the latest commit as this scenario seems vanishingly likely